lib/trento/application/usecases/sap_systems/health_summary_service.ex

```diff
@@ -48,10 +48,26 @@ defmodule Trento.SapSystems.HealthSummaryService do
       sapsystem_health: health,
       database_health: compute_database_health(database_instances),
       clusters_health: compute_clusters_health(all_instances),
-      hosts_health: compute_hosts_health(all_instances)
+      hosts_health: compute_hosts_health(all_instances),
+      hana_cluster_id: extract_hana_cluster_id(database_instances),
+      database_id: extract_database_id(database_instances)
     })
   end
 
+  @spec extract_database_id([DatabaseInstanceReadModel.t()]) :: String.t()
+  defp extract_database_id(database_instances) do
+    database_instances
+    |> Enum.map(fn %{sap_system_id: database_id} -> database_id end)
```

Comment:
We don' need to iterate all over the databases. Maybe simply:
```
database_instances
|> List.first()
|> Map.get(sap_system_id)
```
```

Reply:
pay attention here since if the map is empty `List.first()` returns nil

Reply:
Or better

```elixir
  def extract_database_id([]), do: nil
  def extract_database_id([%DatabaseInstance{sap_system_id: sap_system_id} | _]), do: sap_system_id
```

Reply:
Done with the match clause as @fabriziosestito suggested, thanks guys